### DISCUSSION TITLE
Terminate worker using threadId

### DISCUSSION THREAD

**Discussion by amansanghvi** [nodejs/help#4867] [Ideas]
Hey, I can't find any current examples of it working, and I don't think it's implemented, but I think it would be really useful to be able to do a lookup of workers using a threadId.

Currently if you want to kill a worker you have to hold the worker instance and call terminate(), or have some sort of shared memory. I think because we have a threadId, it would be good to have a way to kill a worker using that Id. Thoughts?

This could also be expanded to having a lookup for a worker using the threadId so we don't have to pass around an object and can pass a threadId instead.

**Comment by jasnell** [comment]
I'm not sure what version it has been back ported to, but it would be fairly simple to implement this using the new `process.on('worker')` event. It does require holding the reference to the worker instance but it would be fairly easy to implement in a dedicated module.
